src/app/devices/components/DeviceHeaderForms/DeviceHeaderForms.test.tsx

```diff
@@ -1,31 +1,50 @@
-import { mount } from "enzyme";
-import { Provider } from "react-redux";
-import { MemoryRouter } from "react-router-dom";
+import { screen } from "@testing-library/react";
 import configureStore from "redux-mock-store";
 
 import DeviceHeaderForms from "./DeviceHeaderForms";
 
 import { DeviceHeaderViews } from "app/devices/constants";
-import { rootState as rootStateFactory } from "testing/factories";
+import type { RootState } from "app/store/root/types";
+import {
+  domain as domainFactory,
+  domainState as domainStateFactory,
+  rootState as rootStateFactory,
+  subnet as subnetFactory,
+  subnetState as subnetStateFactory,
+  zone as zoneFactory,
+  zoneGenericActions as zoneGenericActionsFactory,
+  zoneState as zoneStateFactory,
+} from "testing/factories";
+import { renderWithBrowserRouter } from "testing/utils";
 
-const mockStore = configureStore();
+const mockStore = configureStore<RootState>();
 
 describe("DeviceHeaderForms", () => {
   it("can render the Add Device form", () => {
-    const state = rootStateFactory();
+    const state = rootStateFactory({
+      domain: domainStateFactory({
+        items: [domainFactory({ id: 0, name: "maas" })],
+        loaded: true,
+      }),
+      subnet: subnetStateFactory({
+        items: [subnetFactory({ id: 0, name: "subnet" })],
+        loaded: true,
+      }),
+      zone: zoneStateFactory({
+        genericActions: zoneGenericActionsFactory({ fetch: "success" }),
+        items: [zoneFactory({ id: 0, name: "default" })],
+      }),
+    });
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter>
-          <DeviceHeaderForms
-            devices={[]}
-            headerContent={{ view: DeviceHeaderViews.ADD_DEVICE }}
-            setHeaderContent={jest.fn()}
-          />
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <DeviceHeaderForms
+        devices={[]}
+        headerContent={{ view: DeviceHeaderViews.ADD_DEVICE }}
+        setHeaderContent={jest.fn()}
+      />,
+      { store }
     );
 
-    expect(wrapper.find("AddDeviceForm").exists()).toBe(true);
+    expect(screen.getByLabelText("Add device")).toBeInTheDocument();
```

Comment:
This returns the "Add device" button, to get the form you'd need to use `screen.getByRole("form", { name: "Add device" })`